### ISSUE TITLE
HTML preview for tables is not accurate to MD language

### ISSUE THREAD

**Issue by fabienrenaud** [vsch/idea-multimarkdown#261]
I wrote some MD in IntelliJ with the Mark Navigator plugin and used the HTML preview to see my tables. They looked good in IntelliJ and didn't display as tables at all on github.com

Bad code what showed good previews:

    | A | B | C |
    |-----------|
    | a | b | c |
    | b | a | c |

and

    | A | B | C |
    | a | b | c |
    | b | a | c |

Displays on github something like:

    | A | B | C | |-----------| | a | b | c | | b | a | c |

According to the [MD cheatsheet](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#tables), MD expects:

    | A | B | C |
    |---|---|---|
    | a | b | c |
    | b | a | c |

Proof: https://github.com/fabienrenaud/java-json-benchmark/blob/339219d2283c9a72fd0119310032d2cf54ef376a/README.md

**Comment by vsch** [comment]
@fabienrenaud, pegdown parser uses PhpExtra table extension format which has more flexibility than GitHub. It allows for disparate number of columns in the header, separator and body of the table.

As long as there is at least one column in the separator row and one header or one body row then it will recognize it as a table. It will also allow multiple header rows, which GitHub cannot handle.

In addition Markdown Navigator has to have a few mods to the language to allow for enhanced features. If parsing would allow only GitHub table limitations it would be impossible to make format as you type editing feature.

Formatting a table or enabling format as you type in code style settings of the enhanced plugin would automatically fix the table to:

```markdown
| A | B | C |
|---|---|---|
| a | b | c |
| b | a | c |
```

![tableaddmissing](https://cloud.githubusercontent.com/assets/10299086/16394691/77b99874-3c84-11e6-90be-0cc51ab691c0.gif)
![tableautoformat](https://cloud.githubusercontent.com/assets/10299086/16394692/77b9b26e-3c84-11e6-8008-856a61ee5fb0.gif)

**Comment by fabienrenaud** [comment]
Hmm. Ok. Thank you for the quick reply.
Won't fix i suppose then?

**Comment by vsch** [comment]
@fabienrenaud, I can address it in two ways:

1. Add an option in HTML rendering settings to render invalid GitHub tables as text. This I can add to the open source plugin since it does not require a full parser to be present, which is only available in the enhanced version.

2. Add an inspection to warn when a table is not in a format GitHub will recognize and a quick fix to fix it. This can only be added to enhanced edition since it requires a full parser.

I will probably do both, but only after I have the pegdown replacement parser complete. 

However, auto-formatting and format document or format element on a table will fix the issue and not require any effort from the user.

**Comment by vsch** [comment]
@fabienrenaud, just released EAP 1.9.9.24 with GFM rendering option. Parser uses non-GFM rules so that enhanced features are not broken, HTML rendering uses GFM rules (optional) so HTML looks like github. Also added inspection and quick fix.

<https://github.com/vsch/idea-multimarkdown/raw/master/dist/idea-multimarkdown.1.9.9.24.zip>

<img width="684" alt="screen shot 2016-07-30 at 1 35 24 pm" src="https://cloud.githubusercontent.com/assets/10299086/17272227/c5e1f64c-565e-11e6-971a-2165b38a8245.png">
